### ISSUE TITLE
Crash when using .helper API in Elasticearch client

### ISSUE THREAD

**Issue by bartosz347** [newrelic/node-newrelic#1908]
## Description

The following error is thrown when making a call to ElasticSearch when NewRelic is enabled. Please refer to sections below for steps to reproduce and software versions. When NewRelic is not loaded, this error does not occur. 

```
TypeError: Cannot read properties of null (reading 'then')
at Object.then (/app/node_modules/@elastic/elasticsearch/lib/Transport.js:156:18)
at DatastoreShim.interceptPromise (/app/node_modules/newrelic/lib/shim/shim.js:1713:8)
at DatastoreShim.bindPromise (/app/node_modules/newrelic/lib/shim/shim.js:1729:15)
at Transport._doRecord (/app/node_modules/newrelic/lib/shim/shim.js:937:22)
at Transport.wrapper (/app/node_modules/newrelic/lib/shim/shim.js:887:24)
at Client.bulkApi [as bulk] (/app/node_modules/@elastic/elasticsearch/api/api/bulk.js:67:25)
at tryBulk (/app/node_modules/@elastic/elasticsearch/lib/Helpers.js:691:16)
at bulkOperation (/app/node_modules/@elastic/elasticsearch/lib/Helpers.js:658:7)
at send (/app/node_modules/@elastic/elasticsearch/lib/Helpers.js:629:9)
at iterate (/app/node_modules/@elastic/elasticsearch/lib/Helpers.js:543:9)
at async XXX.YYY (/app/dist/XXX)
at async XXX.ZZZ (/app/dist/XXX)
```

## Expected Behavior

No error or possibility to disable ElasticSearch instrumentation as a workaround.

## Steps to Reproduce

[NOTE]: # ( Please be as specific as possible. )
[TIP]:  # ( Link a sample application that demonstrates the issue. )

Call Elastic search with the following snippet:
```ts
await elasticClient.helpers.bulk({
  datasource: [
    {
      amount: 123,
    },
  ],
  onDocument() {
    return [
      {
        update: { _index: 'xxx', _id: 'yyyyyy' },
      },
      { doc_as_upsert: true },
    ];
  },
});
```

https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/client-helpers.html#_update_2

## Your Environment

 - Node.js v20
 - ElasticSearch client `@elastic/elasticsearch v7.13.0`
 - ElasticSearch server run with Docker image `elasticsearch:7.13.0`
 - NewRelic v11.6.0
 - OS: MacOS Ventura (M1)

Known NewRelic version that has does not cause this issue `v11.1.0`.

## Additional context

Previously reported in https://github.com/newrelic/node-newrelic/issues/1809#issuecomment-1845082104

**Comment by bizob2828** [comment]
@bartosz347 we took some time triaging this and it appears there is a bug in the elastic search helpers API that they fixed in [7.16.0](https://github.com/elastic/elasticsearch-js/pull/1594). Can you try upgrading at least to that version? May I ask why you are stuck on a 3+ year old version of the library?

**Comment by bartosz347** [comment]
Hi @bizob2828, thanks for your efforts. I'd have to investigate it more thoroughly, but there [was a change in v7.14.0](https://github.com/elastic/elasticsearch-js/issues/1519.) that may cause issues with some of our environments.

Is it possible to disable ElasticSearch instrumentation? I'm aware that my ES version is outdated, but ES DB is rather rarely used in the discussed project, so prioritising the upgrade would be difficult at this moment. Possibility to upgrade NewRelic client is more important for me than ES instrumentation.
